packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -290,12 +321,60 @@ class CupertinoNavigationBar extends StatefulWidget implements ObstructingPrefer
     this.transitionBetweenRoutes = true,
     this.heroTag = _defaultHeroTag,
     this.bottom,
-  }) : assert(
+  }) : largeTitle = null,
+       assert(
+         !transitionBetweenRoutes || identical(heroTag, _defaultHeroTag),
+         'Cannot specify a heroTag override if this navigation bar does not '
+         'transition due to transitionBetweenRoutes = false.',
+       );
+
+  /// Creates a static iOS style navigation bar, with a left aligned [largeTitle].
+  ///
+  /// Similar to the expanded state of [CupertinoSliverNavigationBar], which
+  /// can dynamically change size in response to scrolling.
+  ///
+  /// See also:
+  ///
+  ///   * [CupertinoNavigationBar]'s base constructor, which creates a static
+  ///     iOS style navigation bar with [middle], similar to the collapsed state
+  ///     of [CupertinoSliverNavigationBar].
+  const CupertinoNavigationBar.large({
+    super.key,
+    this.largeTitle,
+    this.leading,
+    this.automaticallyImplyLeading = true,
+    bool automaticallyImplyTitle = true,
+    this.previousPageTitle,
+    this.trailing,
+    this.border = _kDefaultNavBarBorder,
+    this.backgroundColor,
+    this.automaticBackgroundVisibility = true,
+    this.enableBackgroundFilterBlur = true,
+    this.brightness,
+    this.padding,
+    this.transitionBetweenRoutes = true,
+    this.heroTag = _defaultHeroTag,
+  }) : middle = null,
+       automaticallyImplyMiddle = automaticallyImplyTitle,
+       assert(
          !transitionBetweenRoutes || identical(heroTag, _defaultHeroTag),
          'Cannot specify a heroTag override if this navigation bar does not '
          'transition due to transitionBetweenRoutes = false.',
        );
 
+  /// The navigation bar's title, when using [CupertinoNavigationBar.large].
+  ///
+  /// The title is displayed similar to the expanded state of
+  /// [CupertinoSliverNavigationBar], which can dynamically change size in
+  /// response to scrolling. For the smaller, or collapsed, version of this static
+  /// navigation bar, use the base [CupertinoNavigationBar] constructor and
+  /// [middle].
+  ///
+  /// If null and [automaticallyImplyMiddle] is true, an appropriate [Text]
+  /// title will be created if the current route is a [CupertinoPageRoute] and
+  /// has a `title`.
```

Comment:
I suggest defining the property independently and moving the "comparison" section to the end, so that the reader don't need to understand the other widget to understand this one.
```suggestion
  /// If null and [automaticallyImplyMiddle] is true, an appropriate [Text]
  /// title will be created if the current route is a [CupertinoPageRoute] and
  /// has a `title`.
  ///
  /// This property is null for the base [CupertinoNavigationBar] constructor,
  /// which shows a collapsed navigation bar and assigns [middle] instead.
  ///
  /// See also:
  ///  * [CupertinoSliverNavigationBar.largeTitle], a similar property 
  ///    in the expanded state of [CupertinoSliverNavigationBar], which can
  ///    dynamically change size in response to scrolling.
```

(If you agree with this one, the doc for `middle` should also be changed.)

Reply:
I am not sure I understand, can you explain more?

Reply:
Sure! I think it would be better to introduce this widget just for itself, instead of introducing it as "similar to the other widget", in which case the developer would need to understand the other widget in order to understand this one.

Reply:
That is a really good point! Thank you for explaining!